### PR TITLE
Robot Driving Backwards

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
A fix to a bug where some robots on some fields would either drive backwards or not straight. The source of this issue was the impulse vector of the collision between the wheel and the ground on some fields being upside down. I fixed this by adding a check in customwheel.cs to see if the y component of the impulse is negative, and if so multiplying it by -1.

### Objectives
- [x] Fix the issue
- [x] Test the new code on several robots across several fields

### Testing Done
- Several robots have been tested on various fields, and they all seem to behave as expected
- Testing with multiple robots at the same time has also been done

[JIRA Issue](https://jira.autodesk.com/browse/AARD-XXXX)

### CODE REVIEW

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]

**Review by a-crowell — APPROVED** [review]
Tested on multiple/duplicate robots and worked every time. Good work!

File: engine/engine.sln.DotSettings.user [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:String x:Key="/Default/CodeInspection/ExcludedFiles/FilesAndFoldersToSkip2/=28740E66_002D4B28_002DD1EE_002D7FFD_002DCA276D12197B_002Fd_003AAssets_002Fd_003AScripts_002Fd_003APhysics_002Ff_003ACustomWheel_002Ecs/@EntryIndexedValue">ForceIncluded</s:String></wpf:ResourceDictionary>
```
[the remark this answers is not in the document]
Comment (KyroVibe) [reply engine/engine.sln.DotSettings.user:2]: @LucaHaverty please add this file to the [.gitignore](/engine/.gitignore) located in engine.

File: engine/Assets/Scripts/Physics/CustomWheel.cs [hunk 2]
```diff
@@ -63,61 +61,63 @@ public class CustomWheel : MonoBehaviour {
...
-    private Vector3 _staticImpulseVecAccum = new Vector3();
-    private Vector3 _rollingImpulseVecAccum = new Vector3();
-    public void OnCollisionStay(Collision collision) {
-        // _collisionCalls++;
-        _pairings.Add((collision.impulse, collision.relativeVelocity));
+    
+    public void OnCollisionStay(Collision collision)
+    {
+        // Flip impulse vector if for whatever reason it is backwards (happens with mean machine robot)
+        Vector3 impulse = (collision.impulse.y < 0f) ? -collision.impulse: collision.impulse;
```
Comment (KyroVibe) [inline engine/Assets/Scripts/Physics/CustomWheel.cs:77]: Change to recalculated impulse vector using point of contact and center of mass as discussed.

**Review by KyroVibe — CHANGES_REQUESTED** [review]

File: engine/engine.sln.DotSettings.user (hunk 1 above)
[the remark this answers is not in the document]
Comment (LucaHaverty) [reply engine/engine.sln.DotSettings.user:2]: Added

File: engine/Assets/Scripts/Physics/CustomWheel.cs (hunk 2 above)
Comment (LucaHaverty) [reply engine/Assets/Scripts/Physics/CustomWheel.cs]: Done

**Review by BrandonPacewic — APPROVED** [review]
Looks good!

**Review by PepperLola — APPROVED** [review]

**Review by KyroVibe — APPROVED** [review]

**Pull request merged**